lib/stdlib/doc/src/lists.xml

```diff
@@ -323,6 +323,28 @@ flatmap(Fun, List1) ->
       </desc>
     </func>
 
+    <func>
+      <name name="foldlwhile" arity="3" since=""/>
+      <fsummary>Fold a function over a list until <anno>halt</anno> is returned</fsummary>
+      <desc>
+        <p>Calls <c><anno>Fun</anno>(<anno>Elem</anno>, <anno>AccIn</anno>)</c>
+          on successive elements <c>A</c> of <c><anno>List</anno></c>, starting
+          with <c><anno>AccIn</anno> == <anno>Acc0</anno></c>.
+          <c><anno>Fun</anno>/2</c> must return a tuple <c>{cont, Acc}</c>
+          to continue or <c>{halt, Acc}</c>
```

Comment:
`cont` and `halt` sound strange, I have never encountered something like that anywhere else, certainly not in the `lists` module.

Reply:
any suggestion?

Reply:
If it is to follow the "style" of the `lists` module, it probably boils down to `true` and `false` 😓 That is not very intuitive I admit, but OTOH the fold function _is_ halfway a predicate, so... Well, what do others think? 😅

Reply:
I take the terminology from Elixir. I think that an atom say a lot more than a simple boolean.

Reply:
Is the same problem that we have with the gen_server or other part of the stdlib.
Remembering stuff is hard but {true, Acc} or {false, Acc} for me doesn't means nothing.

Probably is better `Acc` and `{stop, Acc}`?

Reply:
> Is the same problem that we have with the gen_server or other part of the stdlib.

There it is justified, they are complex, you simply can't get away with booleans there of course ;)

> Remembering stuff is hard but {true, Acc} or {false, Acc} for me doesn't means nothing.

It's obvious from the `foldlwhile` context it is used in IMO (and the `...while` in the function name hints at `while ($some_boolean) {...}` loops as found in other languages), but I may be biased and have just gotten used to it.

> * `Acc` and `{stop, Acc}`

No, this is ambiguous. `Acc` could be a 2-tuple with `stop` as first element.

> * `{true, Acc}` and `{false, Acc}`

That is my favorite :) Btw, using booleans means you can also use any function that evaluates to a boolean there. Compare this fold function...
```erlang
fun
    (N, Acc) -> {N<3, [N|Acc]}
end
```
... with this...
```erlang
fun
    (N, Acc) ->
        ContOrHalt=case N<3 of
            true -> cont;
            false -> halt
        end,
        {ContOrHalt, [N|Acc]}
end
```
... or this...
```erlang
fun
    (N, Acc) when N<3 ->
        {cont, [N|Acc]};
    (N, Acc) ->
        {halt, [N|Acc]}
end
```

Reply:
Yes I see your point but this trick apply only in some simple case like when the discriminator return a boolean. But in all the other case when you have a function that returns `{error, Reason}` you have to fallback use `case ... of` or other helpers